src/components/TextInput/TextInput.js

```diff
@@ -10,11 +10,13 @@ type Props = {
   children?: React$Node,
 };
 
-const TextInput = ({ isFocused, hasError, children, ...delegated }: Props) => (
-  <Wrapper isFocused={isFocused} hasError={hasError}>
-    <InputElem {...delegated} />
-    {children}
-  </Wrapper>
+const TextInput = forwardRef<Props, Wrapper>(
```

Comment:
I went with this approach after skimming [this article](https://medium.com/flow-type/supporting-react-forwardref-and-beyond-f8dd88f35544). To be honest I don't fully understand how the 2 abstract types are used, and I've mostly worked with TS over the last year so I'm a bit rusty on Flow internals. Does it compile without warning with `HTMLElement`?

Reply:
Actually my bad, should be `HTMLDivElement`. But yeah I'm a little rusty as well :) It does compile without warning with `HTMLDivElement`. I also referenced that article and this line is what stood out to me:

> If your component is a built-in, like div or button use the type defined in the dom.js libdef, like HTMLDivElement or HTMLButtonElement.

Tested by making Wrapper return a span, and Flow catches it, so I think it's good?

<img width="600" alt="Screen Shot 2019-04-09 at 8 13 46 PM" src="https://user-images.githubusercontent.com/17421347/55849055-25948d00-5b04-11e9-93c8-3996d0729947.png">

Reply:
Hm, nvm - I think I misunderstood the article and this is not the right approach. Flow doesn't catch if I change the styled component to something other than div, e.g. styled.span. Yours seems like the right approach, what if we just moved `getBorderColor` and `Wrapper` above TextInput to fix the lint warning? 🙃

Reply:
That sounds good to me. I feel like `no-use-before-define` shouldn't trigger on type declarations since I'm fairly certain they're hoisted but I guess this is a weird case where the type declaration is also a component class.